source/common/runtime/runtime_features.cc

```diff
@@ -79,6 +87,48 @@ ABSL_FLAG(uint64_t, re2_max_program_size_warn_level,            // NOLINT
 
 namespace Envoy {
 namespace Runtime {
+namespace {
+
+std::string swapPrefix(std::string name) {
+  return absl::StrReplaceAll(name, {{"envoy_", "envoy."}, {"features_", "features."}});
+}
+
+} // namespace
+
+// This is a singleton class to map Envoy style flag names to absl flags
+class RuntimeFeatures {
+public:
+  RuntimeFeatures();
+
+  // Get the command line flag corresponding to the Envoy style feature name, or
+  // nullptr if it is not a registered flag.
+  absl::CommandLineFlag* getFlag(absl::string_view feature) const {
+    auto it = all_features_.find(feature);
+    if (it == all_features_.end()) {
+      return nullptr;
+    }
+    return it->second;
+  }
+
+private:
+  absl::flat_hash_map<std::string, absl::CommandLineFlag*> all_features_;
+};
+
+using RuntimeFeaturesDefaults = ConstSingleton<RuntimeFeatures>;
+
+RuntimeFeatures::RuntimeFeatures() {
+  absl::flat_hash_map<absl::string_view, absl::CommandLineFlag*> flags = absl::GetAllFlags();
+  for (auto& it : flags) {
+    absl::string_view name = it.second->Name();
+    if ((!absl::StartsWith(name, "envoy_reloadable_features_") &&
+         !absl::StartsWith(name, "envoy_restart_features_")) ||
```

Comment:
Just so I understand since I'm not that familiar with this code: why this filtering? How are flags accessed if not via this?

Reply:
this largely helps us avoid pulling in erroneous flags in google 3, where we may end up with non-envoy command line flags.